sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/HttpConstants.java

```diff
@@ -371,6 +371,10 @@ public static class StatusCodes {
 
         public static final int SERVICE_UNAVAILABLE = 503;
         public static final int INTERNAL_SERVER_ERROR = 500;
+        /**
+         * Used by {@link RequestCancelledException} to indicate a client side request cancellation
+         */
+        public static final int REQUEST_CANCELLED = 499;
```

Comment:
We did discuss it and worth discussing again. 
408s are timeouts - but not client requested timeouts, so didn't use that. 410/0 are overused so much, so I think its better to avoid them here. This was the context of previous discussions.

Reply:
I think 499 isn't any clearer or less confusing. What is the problem with 408 - we use it already even for network time-out in some cases (we already have to distinguish between 408 sent form service vs. client-side). And new status code introduces a new semantic - I don't see enough benefit of introducing a new one here. Let's discuss in todays CRI scrum - and maybe also get .Net folks opinion?

Reply:
So 408 with new substatus code is what we are going ahead with right? 
408 with substatus code CLIENT_OPERATION_TIMEOUT = 10006;

Reply:
@mbhaskar - substatus code should probably be 20008 - and we should ask .Net folks to add it here https://msdata.visualstudio.com/CosmosDB/_git/CosmosDB?path=/Product/Microsoft.Azure.Documents/SharedFiles/StatusCodes.cs&_a=contents&version=GBmaster to avoid getting inconsistency error codes eventually

Reply:
20K series is for Backend and 10K series is for client side
Example: https://github.com/Azure/azure-sdk-for-java/blob/dd78ea3f1f9becf082fc83e4a95958b17f458661/sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/HttpConstants.java#L405

Reply:
Ok so based on the discussion, the substatus code for this will be 20008